app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/PeerGroupHandshake.kt

```diff
@@ -32,7 +44,75 @@ import java.net.InetAddress
  * initiates the file transfer through [SenderDevice] on the sender and using
  * [ReceiverDevice] on the receiver.
  */
-interface PeerGroupHandshake {
+abstract class PeerGroupHandshake(var wifiDirectManager: WifiDirectManager) {
+  lateinit var outputStream: OutputStream
```

Comment:
these shouldn't be made into fields, this has given them an overly broad scope, they should not be accessible to the entire class as they are only needed in the method `exchangeFileTransferMetadata` so it is fine to have them as parameters as they were